File: Data Structures & Algorithms/valid-sudoku/submission-1.java
class Solution {
    public boolean isValidSudoku(char[][] board) {
        boolean ans =true ;
        HashSet <Character>[] Row = new HashSet[9];
        HashSet  <Character>[] Col = new HashSet[9];
        HashSet  <Character>[] square = new HashSet[9];

        for (int i=0;i<9;i++){
            Row[i]=new HashSet<>();
            Col[i]=new HashSet<>();
            square[i]=new HashSet<>();
        }
    
        for (int i=0; i<9;i++){
            for (int j=0; j<9;j++){
                char val=board[i][j];
                if(val!= '.' ){
                    if(Row[i].contains(val))
                        return false;
                    Row[i].add(val);
                    if (Col[j].contains(val))
                        return false;
                    Col[j].add(val);

                    int boxno= (i/3)*3+j/3;

                    if(square[boxno].contains(val))
                        return false;
                    square[boxno].add(val);
                }
                
            }
        }
        return true;
    }
}

File: Data Structures & Algorithms/set-zeroes-in-matrix/submission-1.cpp
class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        int row = matrix.size();
        int col = matrix[0].size();
        bool rowZero= false;
        for(int r=0;r<row ;r++){
            for(int c=0;c<col;c++){
                if(matrix[r][c]==0){
                    matrix[0][c]=0;
                    if(r>0){
                        matrix[r][0]=0;
                    }
                    else{
                        rowZero=true;
                    }
                }
            }
        }
        for(int r=1;r<row;r++){
            for(int c=1;c<col;c++){
                if(matrix[0][c]==0||matrix[r][0]==0){
                    matrix[r][c]=0;
                }
            }
        }
        if(matrix[0][0]==0){
            for(int r=0;r<row ;r++){
                matrix[r][0]=0;
            }
        }
        if(rowZero){
            for(int c=0;c<col;c++){
                matrix[0][c]=0;
            }
        }

    }

};
